modulus/models/graphcast/graph_cast_processor.py

```diff
@@ -177,3 +179,50 @@ def forward(
             )
 
         return efeat, nfeat
+
+
+class GraphCastProcessorGraphTransformer(nn.Module):
+    """Processor block used in GenCast operating on a latent space
+    represented by hierarchy of icosahedral meshes.
+
+    Parameters
+    ----------
+    """
+
+    def __init__(
+        self,
+        attn_mask,
+        processor_layers: int = 16,
+        input_dim_nodes: int = 512,
+        hidden_dim: int = 512,
+    ):
+        super().__init__()
+
+        layers = [
+            te.pytorch.TransformerLayer(
+                hidden_size=input_dim_nodes,
+                ffn_hidden_size=hidden_dim,
+                num_attention_heads=4,
+                layer_number=i + 1,
+                fuse_qkv_params=False,
+            )
+            for i in range(processor_layers)
+        ]
+
+        self.processor_layers = nn.ModuleList(layers)
+        self.hidden_dim = hidden_dim
+        self.attn_mask = torch.tensor(attn_mask, dtype=torch.bool)
+
+    def forward(
+        self,
+        nfeat: Tensor,
+    ) -> Tensor:
+        nfeat = nfeat.unsqueeze(1)
+        for module in self.processor_layers:
+            nfeat = module(
+                nfeat,
+                attention_mask=self.attn_mask.to(nfeat.device),
```

Comment:
Note: Not sure how efficiently TE implements masking. Need to do some benchmarking.

Reply:
Done, but I'm not sure if it's a good idea to keep the mask in the state_dict as the size of it can be huge.

Reply:
What I meant is call `register_buffer` with the mask tensor and set `persistent=False`. This way, it is not part of the state_dict when storing the model and always created when the module is instantiated but comes with the advantage of `to` calls being propagated. 
But actually, does TE have the requirement of keeping the mask in fp32? Otherwise, don't do that because then the dtype might be affected by setting the dtype of the model.

Reply:
Then register_buffer should work as only floating point buffers are changed when e.g. calling `model.to(dtype=torch.bfloat16)`, we sometimes use this behavior as a workaround to avoid buffers being casted by bitcasting them to int and later bitcasting back. 

I prefer the buffer approach if applicable. Thus, e.g. `model.to(device="cpu")` would move this large mask back to the CPU and not leave a "ghost" buffer back on the device.